include/boost/geometry/algorithms/detail/envelope/linestring.hpp

```diff
@@ -0,0 +1,221 @@
+    static inline OutputIterator push_interval(Longitude const& lon1,
+                                               Longitude const& lon2,
+                                               OutputIterator oit)
+    {
+        typedef longitude_interval<Longitude> interval_type;
+
+        typedef math::detail::constants_on_spheroid
+            <
+                Longitude, radian
+            > constants;
+
+        BOOST_ASSERT(! math::larger(lon1, lon2));
+        BOOST_ASSERT(! math::larger(lon1, constants::max_longitude()));
+
+        if (math::larger(lon2, constants::max_longitude()))
+        {
+            *oit++ = interval_type(lon1, constants::max_longitude());
+            *oit++ = interval_type(constants::min_longitude(),
+                                   lon2 - constants::period());
+        }
+        else
+        {
+            *oit++ = interval_type(lon1, lon2);
+        }
+        return oit;
+    }
+
+    template <typename Linear, typename Box>
+    static inline void apply(Linear const& linear, Box& mbr)
+    {
+        typedef typename coordinate_type<Box>::type box_coordinate_type;
+        typedef typename fp_coordinate_type<Linear>::type calculation_type;
+
+        typedef typename geometry::segment_iterator
+            <
+                Linear const
+            > iterator_type;
+
+        typedef typename coordinate_system
+            <
+                Linear
+            >::type::units linear_units_type;
+
+        BOOST_ASSERT(geometry::num_points(linear) != 0);
+
+        typedef longitude_interval<calculation_type> interval_type;
+
+        calculation_type const pi = math::pi<calculation_type>();
+
+        std::vector<interval_type> longitude_intervals;
+        std::back_insert_iterator
+            <
+                std::vector<interval_type>
+            > oit(longitude_intervals);
+
+        calculation_type lat_min = 0, lat_max = 0;
+        bool first = true;
+        for (iterator_type seg_it = geometry::segments_begin(linear);
+             seg_it != geometry::segments_end(linear);
+             ++seg_it, first = false)
+        {
+            typename point_type<Linear>::type p[2];
+            detail::assign_point_from_index<0>(*seg_it, p[0]);
+            detail::assign_point_from_index<1>(*seg_it, p[1]);
+
+            calculation_type lon1 = geometry::get<0>(p[0]);
+            calculation_type lat1 = geometry::get<1>(p[0]);
+            calculation_type lon2 = geometry::get<0>(p[1]);
+            calculation_type lat2 = geometry::get<1>(p[1]);
+
+            math::normalize_spheroidal_coordinates
+                <
+                    linear_units_type
+                >(lon1, lat1);
+
+            math::normalize_spheroidal_coordinates
+                <
+                    linear_units_type
+                >(lon2, lat2);
+
+            math::convert_coordinates<linear_units_type, radian>(lon1, lat1);
+            math::convert_coordinates<linear_units_type, radian>(lon2, lat2);
```

Comment:
Here we have the same pattern as above (in `intersects_antimeridian`):
1. create variables contain coordinates
2. normalize the coorinates
3. convert to radians (this isn't done above)

I'm guessing that this pattern can be used often. Would it be a good idea to have one well-defined function for this instead of several small ones?

Why does `convert_coordinates` was added? We already have `get_as_radian`. I'm guessing that `normalize_spheroidal_coordinates` are normalized using the native units (radians or degrees). Is it safe? Let's say the coordinates are normalized using degrees and then converted to radians, are the radians normalized? I'm guessing that in some cases this may not hold. I think that first the coordinates should be converted to radians and then normalized. Furthermore I think it should be done this way in all algorithms for consistency. Therefore we probably doesn't need those generic utils you added for conversion, even more, they can be harmful. We should be forced to convert the units always the same way in all functions. Most preferably we could have single one function doing it.